scripts/upgrade

```diff
@@ -7,108 +7,35 @@
 
 upgrade_type=$(ynh_check_app_version_changed)
 
-#=================================================
-# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
-#=================================================
-ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=30
-
-# Backup the current version of the app
-ynh_backup_before_upgrade
-ynh_clean_setup_migration() {
-	yunohost app remove $app
-	app=garradin ynh_restore_upgradebackup
-}
-
-ynh_clean_setup () {
-	# Restore it if the upgrade fails
-	if [[ $migration_process -eq 1 ]]; then
-		ynh_clean_setup_migration
-	else
-		ynh_restore_upgradebackup
-	fi
-}
-# Exit if an error occurs during the execution of the script
-ynh_abort_if_errors
-
 #=================================================
 # STANDARD UPGRADE STEPS
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
 ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
-# If final_path doesn't exist, create it
-if [ -z "$final_path" ]; then
-	final_path=/var/www/$app
-	ynh_app_setting_set --app=$app --key=final_path --value=$final_path
-fi
-
 # If secret_key doesn't exist, create it
-if [[ -z "$secret_key" ]]; then
+if [[ -z "${secret_key:-}" ]]; then
 	secret_key=$(ynh_string_random --length=50)
 	ynh_app_setting_set --app=$app --key=secret_key --value=$secret_key
 fi
 
-
-# Cleaning legacy permissions
-if ynh_legacy_permissions_exists; then
-	ynh_legacy_permissions_delete_all
-
-	ynh_app_setting_delete --app=$app --key=is_public
-fi
-
-#=================================================
-# MIGRATION FROM Garradin
-#=================================================
-
-[[ $YNH_APP_ID == "garradin" ]] \
-	&& [[ "$(cat "$final_path/VERSION")" != 1.2.4 ]] \
-	&& ynh_die --message "It looks like that you have an old Garradin installed. You need to first upgrade your Garradin instance (id: $garradin_migrate_id) and after to migrate to Paheko."
-
-ynh_handle_app_migration --migration_id=garradin --migration_list=garradin_migrations
-
-if [[ $migration_process -eq 1 ]]; then
-	final_path=$(ynh_app_setting_get --app=$app --key=final_path)
-	ynh_permission_update --permission="main" --add="all_users"
-	ynh_permission_update --permission="main" --add="visitors"
-fi
-
-#=================================================
-# STANDARD UPGRADE STEPS
-#=================================================
-
 #=================================================
-# CREATE DEDICATED USER
+# BACKUP SQLITE DATABASE
 #=================================================
-ynh_script_progression --message="Making sure dedicated system user exists..." --weight=1
+ynh_script_progression --message="Backup of the DB in temporary file...." --weight=5
 
-# Create a dedicated user (if not existing)
-ynh_system_user_create --username=$app --home_dir="$final_path"
+# keep a data directory backup
+cp -a $install_dir/data /tmp/data
```

Comment:
That part aint great from a security point of view, this could accidentally allow access to any user of the machine if permissions are not set properly ... Is this really needed considering the `data` folder is listed in `--keep` in ynh_setup_source later ?

Reply:
Ho, I explain you my problems... 
I have try with the line as before
 `ynh_setup_source --dest_dir="$install_dir" --keep="association.sqlite data config.local.user.php"`,  
but like this there was a strange folder duplicated `data/data` and some files no more needed. The developper of the  app told me something was wrong in the process of upgrade.  Normally the correct way for upgrading is something like (in french):

> pour éviter que des anciens fichiers traînent le mieux est de :
> - décompresser la nouvelle archive dans un répertoire vierge
> - y recopier le répertoire data uniquement
> - vérifier que tout marche bien
> - pointer le virtualhost sur ce nouveau répertoire (ou renommer le
>   nouveau répertoire à la place de l'ancien, peu importe).

So I have next added this suggestion from someone in the tchat:
`ynh_setup_source --dest_dir="$install_dir" --keep="association.sqlite data config.local.user.php" --full_replace=1` ,
but doing like will not keep the file for DB, association.sqlite. It is better beacause it stays like a fresh install and remove old folders no more needed.

Finally I decide try stay with this line, but backup the folder `data` before to copy it again as suggested by the developper...

Perhaps there is another way to do this ?

Reply:
> That part aint great from a security point of view, this could accidentally allow access to any user of the machine if permissions are not set properly ... Is this really needed considering the `data` folder is listed in `--keep` in ynh_setup_source later ?

If I stay with this process  with `cp` and later `rsync`, the folder `data`  is not needed in the `--keep`

Reply:
You told me if I do so it is better comment or remove
`ynh_system_user_create --username=$app --home_dir="$final_path"` ??
Is this possible to do ??

Reply:
Why not use `data_dir` (`/home/yunohost.app/paheko`) to store all data so it is not affected during upgrade?